app/main.py

```diff
@@ -1,34 +1,230 @@
+    def fire(self, row: int, column: int) -> None:
         # Change the `is_alive` status of the deck
         # And update the `is_drowned` value if it's needed
-        pass
+        """
+        This method exchanges state decks or ships
+        :param row:
+        :param column:
+        :return:
+        """
+        decks = self.decks
+        for deck in decks:
+            if deck.row == row and deck.column == column:
+                deck.is_alive = False
+                deck.symbol = "*"
+        decks_values = [deck.is_alive for deck in self.decks]
+        if not any(decks_values):
+            self.is_drowned = True
+            for deck in self.decks:
+                deck.symbol = "x"
+
+    @staticmethod
+    def create_decks(start: tuple, end: tuple) -> list[Deck]:
+        """
+        This metod creates ship's decks
+        :param start:
+        :param end:
+        :return:
+        """
+        result = []
+        row_start = start[0]
+        row_end = end[0]
+        column_start = start[1]
+        column_end = end[1]
+        if start == end:
+            result.append(Deck(row_start, column_start))
+            return result
+        if row_start == row_end:
+            for i in range(column_start, column_end + 1):
+                result.append(Deck(row_start, i))
+        if column_start == column_end:
+            for i in range(row_start, row_end + 1):
+                result.append(Deck(i, column_start))
+        return result
+
+    def __repr__(self) -> str:
+        return str(self.__dict__)
 
 
 class Battleship:
-    def __init__(self, ships):
+    def __init__(self, ships: list[tuple]) -> None:
         # Create a dict `self.field`.
         # Its keys are tuples - the coordinates of the non-empty cells,
         # A value for each cell is a reference to the ship
         # which is located in it
-        pass
+        self.ships = {coords: Ship(coords[0], coords[1]) for coords in ships}
+        self.field = [
+            (row, column)
+            for row in range(10)
+            for column in range(10)
+        ]
+        self._validate_field()
 
-    def fire(self, location: tuple):
+    def fire(self, location: tuple) -> str:
         # This function should check whether the location
         # is a key in the `self.field`
         # If it is, then it should check if this cell is the last alive
         # in the ship or not.
-        pass
+        """
+        This method fires at the ship for transmitted location
+        :param location:
+        :return:
+        """
+
+        current_ship = None
+
+        for ship in self.ships.values():
+            if ship.is_drowned:
+                continue
+            deck = ship.get_deck(location[0], location[1])
+            if deck:
+                current_ship = ship
+
+        if current_ship and not current_ship.is_drowned:
+            return "Hit!"
+        if current_ship and current_ship.is_drowned:
+            return "Sunk!"
```

Comment:
After hitting a ship's deck, you should check whether all decks of that ship are "sunk" (i.e., is_alive = False) to return "Sunk!".